### PR TITLE
[onert] Apply TrainableBackendContext to `train` backend

### PR DESCRIPTION
**Pull request by ragmani**
This commit makes `train` backend's context trainable.

ONE-DCO-1.0-Signed-off-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: runtime/onert/backend/train/BackendContext.cc [hunk 1]
```diff
@@ -28,34 +28,75 @@ namespace backend
...
+      return;
+    // NOTE Assuming there is no layout changes (Always assume NHWC or UNKNOWN)
+    assert(tgraph.layout() != ir::Layout::NCHW);
 
-  for (auto op_ind : _data.op_order)
-  {
-    auto fn_seq = kernel_gen->generate(op_ind);
-    ret.emplace_back(op_ind, std::move(fn_seq));
-  }
+    // TODO Register TensorInfo that has gradient's shape
```
Comment (ragmani) [inline runtime/onert/backend/train/BackendContext.cc:57]: For this, I have a plan to introduce `StaticGradientShapeInfer` (draft #10945)

File: runtime/onert/backend/train/BackendContext.cc [hunk 2]
```diff
@@ -28,34 +28,75 @@ namespace backend
...
+  // TODO Enable
+  // for (auto &&it : ret)
+  // {
+  //   auto &fn_seq = it.second;
+  //   fn_seq->iterate([&](exec::IFunction &ifunc) { ifunc.prepare(); });
+  // }
 
   return ret;
 }
-
```
Comment (jyoungyun) [inline runtime/onert/backend/train/BackendContext.cc:58]: Unnecessary changes. :(

File: runtime/onert/backend/train/KernelGenerator.cc [hunk 3]
```diff
@@ -45,33 +45,19 @@ ops::PoolType convertPoolType(ir::operation::Pool2D::PoolType type_ir)
...
-  ret->enableDynamicShapeInferer(false);
+  // TODO Generate TrainableFnSequence that can go backward as well
+  auto ret = std::make_unique<exec::train::TrainableFnSequence>();
 
-  const auto &op = _graph.operations().at(ind);
-  op.accept(*this);
-  ret->append(releaseFunction());
+  const auto &op = _tgraph.operations().at(ind);
+  // op.accept(*this);
+  // ret->append(releaseFunction());
```
Comment (jyoungyun) [inline runtime/onert/backend/train/KernelGenerator.cc:60]: Why do you comment out this code?

File: runtime/onert/backend/train/KernelGenerator.cc [hunk 4]
```diff
@@ -90,80 +75,6 @@ std::unique_ptr<exec::FunctionSequence> InferenceKernelGenerator::generate(ir::O
...
-
-  auto fn = std::make_unique<ops::PoolLayer>();
-
-  fn->configure(ifm_tensor, padding.left, padding.right, padding.top, padding.bottom,
-                stride.horizontal, stride.vertical, kw, kh, activation, ofm_tensor,
-                convertPoolType(node.param().op_type));
-
-  _return_fn = std::move(fn);
-}
-
```
Comment (jyoungyun) [inline runtime/onert/backend/train/KernelGenerator.cc:166]: I will add this code again after creating trainable operations.

File: runtime/onert/backend/train/KernelGenerator.cc [hunk 5]
```diff
@@ -174,34 +85,10 @@ KernelGenerator::KernelGenerator(const ir::train::TrainableGraph &tgraph,
...
-      tensor->increase_ref();
-    }
-  }
-  return ret;
-}
-
 void KernelGenerator::visit(const ir::train::operation::Loss &)
 {
   // TODO Generate kernel
+  UNUSED_RELEASE(convertPoolType);
```
Comment (jyoungyun) [inline runtime/onert/backend/train/KernelGenerator.cc:91]: What does this code mean?

Comment (ragmani) [reply runtime/onert/backend/train/KernelGenerator.cc:91]: It was a temporary way to keep `convertPoolType`.

**Review by jyoungyun — APPROVED** [review]
LGTM

**Review by hseok-oh — APPROVED** [review]
